provisioner/converge/provisioner.go

```
@@ -0,0 +1,262 @@
+	Params    map[string]string `mapstucture:"params"`
+}
+
+// Provisioner for Converge
+type Provisioner struct {
+	config Config
+}
+
+// Prepare provisioner somehow. TODO: actual docs
+func (p *Provisioner) Prepare(raws ...interface{}) error {
+	err := config.Decode(
+		&p.config,
+		&config.DecodeOpts{
+			Interpolate:        true,
+			InterpolateContext: &p.config.ctx,
+		},
+		raws...,
+	)
+	if err != nil {
+		return err
+	}
+
+	// validate version
+	if !versionRegex.Match([]byte(p.config.Version)) {
+		return fmt.Errorf("Invalid Converge version %q specified. Valid versions include only letters, numbers, dots, and dashes", p.config.Version)
+	}
+
+	// validate sources and destinations
+	for i, dir := range p.config.ModuleDirs {
+		if dir.Source == "" {
+			return fmt.Errorf("Source (\"source\" key) is required in Converge module dir #%d", i)
+		}
+		if dir.Destination == "" {
+			return fmt.Errorf("Destination (\"destination\" key) is required in Converge module dir #%d", i)
+		}
+	}
+
+	// validate modules
+	if len(p.config.Modules) == 0 {
+		return errors.New("Converge requires at least one module (\"modules\" key) to provision the system")
+	}
+	for i, module := range p.config.Modules {
+		if module.Module == "" {
+			return fmt.Errorf("Module (\"module\" key) is required in Converge module #%d", i)
+		}
+		if module.Directory == "" {
+			p.config.Modules[i].Directory = "/tmp"
+		}
+	}
+
+	return err
+}
+
+// Provision node somehow. TODO: actual docs
+func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
+	ui.Say("Provisioning with Converge")
+
+	// bootstrapping
+	if err := p.maybeBootstrap(ui, comm); err != nil {
+		return err // error messages are already user-friendly
+	}
+
+	// check version (really, this make sure that Converge is installed before we try to run it)
+	if err := p.checkVersion(ui, comm); err != nil {
+		return err // error messages are already user-friendly
+	}
```

Let it fail if it's not on the path.